docs/tasks/timeout-injection.md

```diff
@@ -0,0 +1,116 @@
+
+For this tutorial an NGINX front-end will talk to httpbin.
+
+Because the httpbin service we created is not exposed to the outside world for security reasons
+we cannot test it from outside the cluster.  To verify that it is working correctly, we can run
+a _curl_ command against httpbin:8000 *from inside the cluster* using the public _dockerqa/curl_
+image from the Docker hub:
+
+```
+kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/html
+kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/status/500
+time kubectl run -i --rm --restart=Never dummy --image=dockerqa/curl:ubuntu-trusty --command -- curl --silent httpbin:8000/delay/5
+```
+
+Next we will start NGINX and configure it to proxy for the httpbin service started in the previous step
+
+```
+kubectl create -f <(istioctl kube-inject -f nginx-httpbin.yaml)
+```
+
+To test from outside the cluster we will need the IP address of a Kubernetes node, as well
+as the exposed NodePort port on that node.
+
+```
+# Get the IP address of the Kubernetes node
+IP=$(kubectl get po -l infra=istio-ingress-controller -o jsonpath={.items[0].status.hostIP})
+NGINX_NODEPORT=$(kubectl get service nginx --output jsonpath='{.spec.ports[0].nodePort}')
+curl -i $IP:$NGINX_NODEPORT/html
+curl -i $IP:$NGINX_NODEPORT/status/500
+time curl -i $IP:$NGINX_NODEPORT/delay/5
+```
+
+At this point we have two services.  Both are wired through the Istio service mesh.  Both output the same data.
+
+## Manipulating the service mesh with route rules and destination polices
+
+Istio has many advanced capabilities to control how network traffic flows between microservices.  One of the
+most simple capabilities is to add a timeout.
+
+```
+# First, create a rule file
+cat <<EOF > /tmp/httpbin-3s-rule.yaml
+type: route-rule
+name: httpbin-3s-rule
+spec:
+  destination: httpbin.default.svc.cluster.local
+  http_req_timeout:
+    simple_timeout:
+      timeout_seconds: 3
+EOF
+
+# Give the file to istioctl
+istioctl create -f /tmp/httpbin-3s-rule.yaml
+```
+
+Wait a few seconds, then issue `time curl -i $IP:$NGINX_NODEPORT/delay/5` again.  This time the response appears after
+three seconds.  Although _httpbin_ was waiting 5 seconds, Istio cut off the request at 3 seconds.
+
+## Cleanup
+
+To remove the rules, deployments and services used in this tutorial:
+
+```
+istioctl delete route-rule httpbin-3s-rule
+kubectl delete -f httpbin.yaml
+kubectl delete -f nginx-httpbin.yaml
+```
+
+
+## Discussion
+
+We have seen two web services, httpbin and NGINX, connected via an Istio proxy mesh.  We have
+used the proxy mesh to introduce a maximum delay using a timeout in all communications to httpbin.
+
+## Next Steps
```

Comment:
See here for instructions on how to use the templates:

https://istio.io/docs/home/contribute/using-page-templates.html